pkg/controller/datavolume-controller.go

```diff
@@ -373,194 +383,319 @@ func (r *DatavolumeReconciler) Reconcile(_ context.Context, req reconcile.Reques
 		return reconcile.Result{}, err
 	}
 
-	cloneStrategy, err := r.getCloneStrategy()
+	cloneStrategy, err := r.selectCloneStrategy(datavolume, pvcSpec)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
-	bindingMode, err := r.getStorageClassBindingMode(pvcSpec.StorageClassName)
-	if err != nil {
-		return reconcile.Result{}, err
+	if !pvcExists {
+		if cloneStrategy == SmartClone {
+			snapshotClassName, _ := r.getSnapshotClassForSmartClone(datavolume, pvcSpec)
+			return r.reconcileSmartClonePvc(log, datavolume, pvcSpec, transferName, snapshotClassName)
+		}
+		if cloneStrategy == CsiClone {
+			// Right not a NO-OP so falls back to Host Assisted - implementation in progress
+			r.log.V(3).Info("CSI-Clone is available")
+		}
+
+		newPvc, err := r.createPvcForDatavolume(log, datavolume, pvcSpec)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		pvc = newPvc
+	} else {
+		//TODO: maybe... when pvc exists, get clone strategy from pvc?
```

Comment:
If the PVC exists, we can only do a host assisted clone right, since both CSI and smart clone require the PVC to not exist.

Reply:
Yes, and I forgot what was my original Idea behind this comment